phylogenetic/rules/prepare_sequences_segments.smk

```diff
@@ -0,0 +1,47 @@
+"""
+This part of the workflow prepares sequences for constructing the segment phylogenetic tree.
+
+REQUIRED INPUTS:
+
+    metadata    = data/metadata.tsv
+    sequences   = data/sequences.fasta
+    segment_reference   = defaults/lassa_{segment}.fasta
+
+OUTPUTS:
+
+    metadata    = data/metadata_{segment}.tsv
+    sequences   = data/sequences_{segment}.fasta
+
+This part of the workflow usually includes the following steps:
+
+    - nextclade: Aligns sequences to the reference segment sequence and extracts the segment sequences to ensure the reference files are valid
+    - augur filter: Filters the sequences and metadata to include only the sequences that were successfully aligned
+See Nextclade's usage docs for these commands for more details.
+"""
+
+rule prepare_sequences_segments:
+    input:
+        metadata = "data/metadata.tsv",
+        sequences = "data/sequences.fasta",
+        segment_reference = "defaults/lassa_{segment}.fasta",
+    output:
+        metadata = "data/metadata_{segment}.tsv",
+        sequences = "data/sequences_{segment}.fasta",
+    params:
+        strain_id_field = config["strain_id_field"],
+        min_seed_cover = 0.01,
+    shell:
+        """
+        nextclade run \
+            --input-ref {input.segment_reference} \
+            --output-fasta {output.sequences} \
+            --min-seed-cover {params.min_seed_cover} \
+            --silent \
+            {input.sequences}
+        
+        augur filter \
+            --sequences {output.sequences} \
+            --metadata {input.metadata} \
+            --metadata-id-columns {params.strain_id_field} \
+            --output-metadata {output.metadata}
+        """
```

Comment:
Is there a particular reason you didn't split this into two separate rules? I think debugging is always easier when there's a clear separation of commands in rules.

Reply:
No longer relevant